source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -62,32 +63,57 @@ void* MaterialXNode::creator()
 MStatus MaterialXNode::initialize()
 {
 	std::cout << "MaterialXNode::initialize" << std::endl;
+
 	MFnTypedAttribute typedAttr;
-	MFnNumericAttribute nAttr;
 	MFnStringData stringData;
 
 	MObject theString = stringData.create();
 	DOCUMENT_ATTRIBUTE = typedAttr.create(DOCUMENT_ATTRIBUTE_LONG_NAME, DOCUMENT_ATTRIBUTE_SHORT_NAME, MFnData::kString, theString);
 	CHECK_MSTATUS(typedAttr.setStorable(true));
 	CHECK_MSTATUS(typedAttr.setReadable(true));
-    CHECK_MSTATUS(typedAttr.setHidden(true));
+	CHECK_MSTATUS(typedAttr.setInternal(true));
+	CHECK_MSTATUS(typedAttr.setCached(false));
+	CHECK_MSTATUS(typedAttr.setHidden(true));
 	CHECK_MSTATUS(addAttribute(DOCUMENT_ATTRIBUTE));
 
 	ELEMENT_ATTRIBUTE = typedAttr.create(ELEMENT_ATTRIBUTE_LONG_NAME, ELEMENT_ATTRIBUTE_SHORT_NAME, MFnData::kString, theString);
 	CHECK_MSTATUS(typedAttr.setStorable(true));
-    CHECK_MSTATUS(typedAttr.setHidden(true));
-    CHECK_MSTATUS(typedAttr.setReadable(true));
+	CHECK_MSTATUS(typedAttr.setReadable(true));
+	CHECK_MSTATUS(typedAttr.setInternal(true));
+	CHECK_MSTATUS(typedAttr.setCached(false));
+	CHECK_MSTATUS(typedAttr.setHidden(true));
 	CHECK_MSTATUS(addAttribute(ELEMENT_ATTRIBUTE));
 
-	OUT_COLOR_ATTRIBUTE = nAttr.createColor(OUT_COLOR_ATTRIBUTE_LONG_NAME, OUT_COLOR_ATTRIBUTE_SHORT_NAME);
-	CHECK_MSTATUS(typedAttr.setStorable(false));
-	CHECK_MSTATUS(typedAttr.setHidden(false));
-	CHECK_MSTATUS(typedAttr.setReadable(true));
-	CHECK_MSTATUS(typedAttr.setWritable(false));
-	CHECK_MSTATUS(addAttribute(OUT_COLOR_ATTRIBUTE));
+	return MS::kSuccess;
+}
+
+void MaterialXNode::createOutputAttr(MDGModifier& mdgModifier)
+{
+	if (materialXData)
+	{
+		MFnNumericAttribute nAttr;
+
+		MString outputName = materialXData->glslFragmentWrapper->getFragmentName().c_str();
+		_outAttr = nAttr.createColor(outputName, outputName);
+		CHECK_MSTATUS(nAttr.setStorable(false));
```

Comment:
I think this needs to be set storable unless you can re-create this at file-load time. One issue of course is if someone hand edits this in the file, but then I would say they are asking for trouble :).